### ISSUE TITLE
Install-VS for Windows 2019 fails due to azcopy being unavailable

### ISSUE THREAD

**Issue by markatky46** [actions/runner-images#6517]
### Description

My pipeline to build Windows 2019 images is failing each time at the same stage of the install-vs.ps1. 

Provisioning with powershell script: /agent/_work/1/s/runner-images/images/win/scripts/Installers/Install-VS.ps1
    azure-arm: Downloading Bootstrapper ...
    azure-arm: Downloading package from: https://aka.ms/vs/16/release/vs_Enterprise.exe to path C:\Users\install\AppData\Local\Temp\vs_Enterprise.exe .
    azure-arm: Package downloaded successfully in 0.75 seconds
    azure-arm: Enable short name support on Windows needed for Xamarin Android AOT, defaults appear to have been changed in Azure VMs
    azure-arm: Starting Install ...
    azure-arm: Something went wrong with the install.
    azure-arm:
    azure-arm: You can troubleshoot the package failures by:
    azure-arm:
    azure-arm:     1. Search for solutions using the search URL below for each package failure
    azure-arm:     2. Modify your selections for the affected workloads or components and then retry the installation
    azure-arm:     3. Remove the product from your machine and then install again
    azure-arm:
    azure-arm: If the issue has already been reported on the Developer Community, you can find solutions or workarounds there. If the issue has not been reported, we encourage you to create a new issue so that other developers will be able to find solutions or workarounds. You can create a new issue from within the Visual Studio Installer in the upper-right hand corner using the "Provide feedback" button.
    azure-arm:
    azure-arm: ================================================================================
    azure-arm:
    azure-arm: Package 'Microsoft.Azure.Storage.AzCopy.Msi,version=6.2.0.35' failed to download from 'https://az837173.vo.msecnd.net/azcopy-6-2-0/MicrosoftAzureStorageTools.msi'.
    azure-arm:     Search URL
    azure-arm:         https://aka.ms/VSSetupErrorReports?q=PackageId=Microsoft.Azure.Storage.AzCopy.Msi;PackageAction=DownloadPackage;ReturnCode=0x80070194
    azure-arm:     Details
    azure-arm:         WebClient download failed: The remote server returned an error: (404) Not Found.
    azure-arm:         Bits download failed: The operation being requested was not performed because the user has not logged on to the network. The specified service does not exist. (Exception from HRESULT: 0x800704DD)
    azure-arm:         WinInet download failed: Function: HttpQueryInfo-ResponseCode, HR: -2147024492, Message: Url 'https://az837173.vo.msecnd.net/azcopy-6-2-0/MicrosoftAzureStorageTools.msi'
    azure-arm:     Impacted workloads
    azure-arm:         Azure development (Microsoft.VisualStudio.Workload.Azure,version=16.11.31503.43)
    azure-arm:     Impacted components
    azure-arm:         Azure Storage AzCopy (Microsoft.VisualStudio.Component.Azure.Storage.AzCopy,version=16.0.28517.75)
    azure-arm:
    azure-arm: Non zero exit code returned by the installation process : 1

On clicking the link i get and error

<Error>
<Code>ResourceNotFound</Code>
<Message>The specified resource does not exist. RequestId:5e87eb1c-a01e-0038-1139-f0555a000000 Time:2022-11-04T10:35:07.8323488Z</Message>
</Error>

### Platforms affected

- [X] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 10.15
- [ ] macOS 11
- [ ] macOS 12
- [X] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

n/a

### Is it regression?

No

### Expected behavior

The expected behaviour is that the build passes this stage and completes successfully.

### Actual behavior

Build fails with the error

==> azure-arm: Provisioning with powershell script: /agent/_work/1/s/runner-images/images/win/scripts/Installers/Install-VS.ps1
    azure-arm: Downloading Bootstrapper ...
    azure-arm: Downloading package from: https://aka.ms/vs/16/release/vs_Enterprise.exe to path C:\Users\install\AppData\Local\Temp\vs_Enterprise.exe .
    azure-arm: Package downloaded successfully in 0.75 seconds
    azure-arm: Enable short name support on Windows needed for Xamarin Android AOT, defaults appear to have been changed in Azure VMs
    azure-arm: Starting Install ...
    azure-arm: Something went wrong with the install.
    azure-arm:
    azure-arm: You can troubleshoot the package failures by:
    azure-arm:
    azure-arm:     1. Search for solutions using the search URL below for each package failure
    azure-arm:     2. Modify your selections for the affected workloads or components and then retry the installation
    azure-arm:     3. Remove the product from your machine and then install again
    azure-arm:
    azure-arm: If the issue has already been reported on the Developer Community, you can find solutions or workarounds there. If the issue has not been reported, we encourage you to create a new issue so that other developers will be able to find solutions or workarounds. You can create a new issue from within the Visual Studio Installer in the upper-right hand corner using the "Provide feedback" button.
    azure-arm:
    azure-arm: ================================================================================
    azure-arm:
    azure-arm: Package 'Microsoft.Azure.Storage.AzCopy.Msi,version=6.2.0.35' failed to download from 'https://az837173.vo.msecnd.net/azcopy-6-2-0/MicrosoftAzureStorageTools.msi'.
    azure-arm:     Search URL
    azure-arm:         https://aka.ms/VSSetupErrorReports?q=PackageId=Microsoft.Azure.Storage.AzCopy.Msi;PackageAction=DownloadPackage;ReturnCode=0x80070194
    azure-arm:     Details
    azure-arm:         WebClient download failed: The remote server returned an error: (404) Not Found.
    azure-arm:         Bits download failed: The operation being requested was not performed because the user has not logged on to the network. The specified service does not exist. (Exception from HRESULT: 0x800704DD)
    azure-arm:         WinInet download failed: Function: HttpQueryInfo-ResponseCode, HR: -2147024492, Message: Url 'https://az837173.vo.msecnd.net/azcopy-6-2-0/MicrosoftAzureStorageTools.msi'
    azure-arm:     Impacted workloads
    azure-arm:         Azure development (Microsoft.VisualStudio.Workload.Azure,version=16.11.31503.43)
    azure-arm:     Impacted components
    azure-arm:         Azure Storage AzCopy (Microsoft.VisualStudio.Component.Azure.Storage.AzCopy,version=16.0.28517.75)
    azure-arm:
    azure-arm: Non zero exit code returned by the installation process : 1

### Repro steps

1) Build an image using Windows 2019 configuration

**Comment by igorboskovic3** [comment]
Hey @markatky46 we will take a look

**Comment by childsc** [comment]
We are getting the same issue. We believe this issue is somehow resulting from a Windows Update, possibly an update from this week. However, we are not 100% but it was working last week and the only change has been the Windows updates.

**Comment by brittandeyoung** [comment]
It looks like the asset is just no longer available, did Microsoft take it down possibly?: 

```
curl https://az837173.vo.msecnd.net/azcopy-6-2-0/MicrosoftAzureStorageTools.msi
<?xml version="1.0" encoding="utf-8"?><Error><Code>ResourceNotFound</Code><Message>The specified resource does not exist.
RequestId:0aaef18d-f01e-0025-04ec-f258e6000000
Time:2022-11-07T21:05:27.4131802Z</Message></Error>

```

**Comment by rajeevoncloud9** [comment]
getting the same error

**Comment by markatky46** [comment]
@brittandeyoung it would appear so but that subcomponent is still cited as available by Microsoft so unsure why it would have been removed.

**Comment by markatky46** [comment]
Sorry slipped and clicked the wrong button!

**Comment by desek** [comment]
A workaround to complete the image build is to remove the AzCopy component from the VS installer at https://github.com/actions/runner-images/blob/main/images/win/toolsets/toolset-2019.json#L304

It's easy to install azcopy in the workflow until this is fixed.

**Comment by brittandeyoung** [comment]
There is an open issue here on the developer community that I have been trying to get information on. I am not sure where the proper location is to report this to Microsoft as being gone. 

https://developercommunity.visualstudio.com/t/PackageId:MicrosoftAzureStorageAzCopy/10188273?f=&type=question+OR+problem&type=question+OR+problem&c=&redirect=search%2Fsearch&sort=relevance&utm_source=vssetup&utm_medium=setuplog&q=PackageId%3DMicrosoft.Azure.Storage.AzCopy.Msi%3BPackageAction%3DDownloadPackage%3BReturnCode%3D0x80070194

**Comment by markatky46** [comment]
@brittandeyoung i have commented on that article also now.

**Comment by markatky46** [comment]
Thanks @al-cheb

**Comment by al-cheb** [comment]
@brittandeyoung, https://az837173.vo.msecnd.net/azcopy-6-2-0/MicrosoftAzureStorageTools.msi should be restored.

**Comment by markatky46** [comment]
Will you be restoring the feature now? @al-cheb

**Comment by al-cheb** [comment]
> Will you be restoring the feature now? @al-cheb

Yep, it was reverted in this https://github.com/actions/runner-images/pull/6548 PR.

**Comment by markatky46** [comment]
Thanks so much @al-cheb
